src/pages/settings/Payments/TransferBalancePage.js

```diff
@@ -166,70 +167,85 @@ class TransferBalancePage extends React.Component {
         const isButtonDisabled = !isTransferable || !selectedAccount;
         const errorMessage = !_.isEmpty(this.props.walletTransfer.errors) ? _.chain(this.props.walletTransfer.errors).values().first().value() : '';
 
+        const shouldShowTransferView =
+            !_.isEmpty(this.props.userWallet) &&
+            PaymentUtils.hasExpensifyPaymentMethod(this.props.cardList, this.props.bankAccountList) &&
+            this.props.userWallet.tierName &&
+            this.props.userWallet.tierName === CONST.WALLET.TIER_NAME.GOLD;
```

Comment:
Could be simplify with `lodashGet(this.props.userWallet, 'tierName', '')`?
```suggestion
lodashGet(this.props.userWallet, 'tierName', '')
```

Reply:
It can simply be `this.props.userWallet.tierName === CONST.WALLET.TIER_NAME.GOLD;`. We know userWallet is an object. If tierName is `undefined` then it doesn't equal gold.

Reply:
Makes sense. No need to validate `this.props.userWallet.tierName` condition as the execution only reaches here when userWallet obj is available.

Reply:
@mollfpr I am not sure here whether for this straightforward check, do we need to go with `lodashGet` or let it be the current way.

Reply:
done